python/cog/server/connection.py

```diff
@@ -19,11 +19,16 @@ def __init__(self, conn: Connection) -> None:
         self.started = False
 
     async def async_init(self) -> None:
+        if self.started:
+            return
         fd = self.wrapped_conn.fileno()
         # mp may have handled something already but let's dup so exit is clean
         dup_fd = os.dup(fd)
         sock = socket.socket(fileno=dup_fd)
-        sock.setblocking(False)
+        # we don't want to see EAGAIN, we'd rather wait
+        # however, perhaps this is wrong and in some cases this could still block terribly
+        # sock.setblocking(False)
+        sock.setblocking(True)
```

Comment:
I'm having trouble understanding this in the context of the commented out code. Are those concerns for `setblocking(True)`? It'd be nice for this comment to provide enough relevant context for someone to pick this up if we need to revisit this behavior.